rust/lance-index/src/scalar/btree.rs

```diff
@@ -1393,14 +1419,731 @@ pub async fn train_btree_index(
+        )
+        .await;
+    }
+
+    // Clean up partition page files
+    for file_name in part_page_files {
+        cleanup_single_file(
+            store,
+            file_name,
+            "part_",
+            "_page_data.lance",
+            "partition page",
+        )
+        .await;
+    }
+}
+
+/// Helper function to clean up a single partition file
+///
+/// Performs safety checks on the filename pattern before attempting deletion.
+async fn cleanup_single_file(
+    store: &Arc<dyn IndexStore>,
+    file_name: &str,
+    expected_prefix: &str,
+    expected_suffix: &str,
+    file_type: &str,
+) {
+    if file_name.starts_with(expected_prefix) && file_name.ends_with(expected_suffix) {
+        match store.delete_index_file(file_name).await {
+            Ok(()) => {
+                debug!("Successfully deleted {} file: {}", file_type, file_name);
+            }
+            Err(e) => {
+                warn!(
+                    "Failed to delete {} file '{}': {}. \
+                    This does not affect the merge operation, but may leave \
+                    partition files that should be cleaned up manually.",
+                    file_type, file_name, e
+                );
+            }
+        }
+    } else {
+        // If the filename doesn't match the expected format, log a warning but don't attempt deletion
+        warn!(
+            "Skipping deletion of file '{}' as it does not match the expected \
+            {} file pattern ({}*{})",
+            file_name, file_type, expected_prefix, expected_suffix
+        );
+    }
+}
+
+pub(crate) fn part_page_data_file_path(partition_id: u64) -> String {
+    format!("part_{}_{}", partition_id, BTREE_PAGES_NAME)
+}
+
+pub(crate) fn part_lookup_file_path(partition_id: u64) -> String {
+    format!("part_{}_{}", partition_id, BTREE_LOOKUP_NAME)
+}
+
+/// Prefetch configuration for partition iterators
+#[derive(Debug, Clone)]
+pub struct PrefetchConfig {
+    /// Number of batches to prefetch ahead (0 means no prefetching)
+    pub prefetch_batches: usize,
+}
+
+impl Default for PrefetchConfig {
+    fn default() -> Self {
+        Self {
+            prefetch_batches: 1,
+        }
+    }
+}
+
+impl PrefetchConfig {
+    /// Set the prefetch batch count
+    pub fn with_prefetch_batch(&self, batch_count: usize) -> Self {
+        Self {
+            prefetch_batches: batch_count,
+        }
+    }
+}
+
+/// Buffer entry for prefetch queue
+#[derive(Debug)]
+struct BufferEntry {
+    batch: RecordBatch,
+    start_row: usize,
+    end_row: usize,
+}
+
+/// Partition iterator for loading partition data in batches with integrated prefetching
+struct PartitionIterator {
+    reader: Arc<dyn IndexReader>,
+    current_batch: Option<RecordBatch>,
+    current_position: usize,
+    rows_read: usize,
+    batch_size: u64,
+    prefetch_buffer: Arc<tokio::sync::Mutex<VecDeque<BufferEntry>>>,
+    prefetch_task: Option<tokio::task::JoinHandle<()>>,
+}
```

Comment:
I don't think any of this prefetch is needed.  The `FileReader` already does its own prefetch.  Instead of making multiple calls to `read_range` just make a single call to `read_stream` and then use `chunk_concat_stream` to force the stream to emit batches of the correct size.  You will need to add `read_stream` to the `IndexReader` trait but I think that is ok.

The `read_stream` method takes in a `batch_readahead` parameter which controls how many batches it will decode in advance.  There is also a configuration which controls how much data we load in advance and this will default to 2GiB.  However, this is set at the index store level and so it should be a shared 2GiB limit no matter how many readers you have open (e.g. it will not load 2GiB per reader) so we should be fine.